content/docs/add-react-to-a-website.md

```diff
@@ -0,0 +1,179 @@
+
+[View the full example source code](https://gist.github.com/gaearon/faa67b76a6c47adbab04f739cba7ceda)
+
+[Download the full example (2KB zipped)](https://gist.github.com/gaearon/faa67b76a6c47adbab04f739cba7ceda/archive/9d0dd0ee941fea05fd1357502e5aa348abb84c12.zip)
+
+>Note
+>
+>This strategy is mostly useful while React-powered parts of the page are isolated from each other. Inside React code, it's easier to use [component composition](/docs/components-and-props.html#composing-components) instead.
+
+### Tip: Minify JavaScript for Production
+
+Before deploying your website to production, be mindful that unminifed JavaScript can significantly slow down the page for your users.
+
+If you already minify the application scripts, all you need to do is to ensure that the deployed HTML loads the versions of React ending in `production.min.js`:
+
+```js
+<script src="https://unpkg.com/react@16/umd/react.production.min.js" crossorigin></script>
+<script src="https://unpkg.com/react-dom@16/umd/react-dom.production.min.js" crossorigin></script>
+```
+
+If you don't have a minification step for your scripts, [here's one way to set it up](https://gist.github.com/gaearon/42a2ffa41b8319948f9be4076286e1f3).
+
+## Optional: Try React with JSX
+
+In the examples above, we only relied on features that are natively supported by the browsers. This is why we used a JavaScript function call to tell React what to display:
+
+```js
+const e = React.createElement;
+
+// Display a "Like" button
+return e(
+  'button',
+  { onClick: () => this.setState({ liked: true }) },
+  'Like'
+);
+```
+
+However, React also offers an option to use [JSX](/docs/introducing-jsx.html) instead:
+
+```js
+// Display a "Like" button
+return (
+  <button onClick={() => this.setState({ liked: true })}>
+    Like
+  </button>
+);
+```
+
+These two code snippets are equivalent. While **JSX is [completely optional](/docs/react-without-jsx.html)**, many people find it helpful for writing UI code -- both with React and with other libraries.
+
+You can play with JSX using [this online converter](http://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=Q&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2%2Cstage-3&prettier=true&targets=Node-6.12&version=6.26.0&envVersion=).
+
+### Add JSX to a Project
+
+If you want to add JSX to your project, follow these three steps:
+
+1. [Install Node.js](https://nodejs.org/)
+2. **Don't miss this step:** Run `npm init -y` in your project folder
+3. Run `npm install babel-cli@6 babel-preset-react-app@3`
+
+You can now use JSX!
+
+### Compile JSX with One Command
+
+Create create a folder called `src` and run this terminal command:
+
+```
+npx babel --watch src --out-dir . --presets react-app/prod 
```

Comment:
why `.`? feels kinda weird to me for the src and out trees to not be disjoint

Reply:
Because then we don't have to mention that they need to update path in HTML and in Uglify command they learned in previous section. I figured people who would feel it's weird are the ones who'd guess to change the output directory, and the rest would break their setups.